ir.ts

```diff
@@ -28,7 +28,7 @@ export type Expr<A> =
   | {  a?: A, tag: "builtin1", name: string, arg: Value<A> }
   | {  a?: A, tag: "builtin2", name: string, left: Value<A>, right: Value<A>}
   | {  a?: A, tag: "call", name: string, arguments: Array<Value<A>> } 
-
+  | {  a?: A, tag: "call_indirect", fn: Value<A>, arguments: Array<Value<A>> } 
```

Comment:
Connected to my other comment in `compiler.ts`, I like this definition of `call_indirect`! I just think in `lower.ts` you should add any extra args and do the appropriate load ahead of time.